TM1py/Services/CellService.py

```diff
@@ -847,47 +931,88 @@ def extract_cellset_raw(
 
         filter_axis = "$filter=Ordinal ne 2;" if skip_contexts else ""
 
-        url = "/api/v1/Cellsets('{cellset_id}')?$expand=" \
-              "Cube($select=Name;$expand=Dimensions($select=Name))," \
-              "Axes({filter_axis}$expand=Tuples($expand=Members({select_member_properties}" \
-              "{expand_elem_properties}){top_rows}))," \
-              "Cells($select={cell_properties}{top_cells}{skip_cells})" \
-            .format(cellset_id=cellset_id,
-                    top_rows=f";$top={top}" if top and not skip else "",
-                    cell_properties=",".join(cell_properties),
-                    filter_axis=filter_axis,
-                    select_member_properties=select_member_properties,
-                    expand_elem_properties=expand_elem_properties,
-                    top_cells=f";$top={top}" if top else "",
-                    skip_cells=f";$skip={skip}" if skip else "")
+        url=""
+        if sandbox_name == None:
+            url = "/api/v1/Cellsets('{cellset_id}')?$expand=" \
+                "Cube($select=Name;$expand=Dimensions($select=Name))," \
+                "Axes({filter_axis}$expand=Tuples($expand=Members({select_member_properties}" \
+                "{expand_elem_properties}){top_rows}))," \
+                "Cells($select={cell_properties}{top_cells}{skip_cells})" \
+                .format(cellset_id=cellset_id,
+                        top_rows=f";$top={top}" if top and not skip else "",
+                        cell_properties=",".join(cell_properties),
+                        filter_axis=filter_axis,
+                        select_member_properties=select_member_properties,
+                        expand_elem_properties=expand_elem_properties,
+                        top_cells=f";$top={top}" if top else "",
+                        skip_cells=f";$skip={skip}" if skip else "")
+        else:
+            url = "/api/v1/Cellsets('{cellset_id}')?$expand=" \
+                "Cube($select=Name;$expand=Dimensions($select=Name))," \
+                "Axes({filter_axis}$expand=Tuples($expand=Members({select_member_properties}" \
+                "{expand_elem_properties}){top_rows}))," \
+                "Cells($select={cell_properties}{top_cells}{skip_cells})" \
+                "&!sandbox={sandbox_name}".format(cellset_id=cellset_id,
+                        top_rows=f";$top={top}" if top and not skip else "",
+                        cell_properties=",".join(cell_properties),
+                        filter_axis=filter_axis,
+                        select_member_properties=select_member_properties,
+                        expand_elem_properties=expand_elem_properties,
+                        top_cells=f";$top={top}" if top else "",
+                        skip_cells=f";$skip={skip}" if skip else "",
+                        sandbox_name=format_url(sandbox_name))
```

Comment:
I think it's a legacy issue, but shouldn't we be using the `format_url` here instead of calling the `format` function on the string?
If we do enhance the `format_url` function to support sandboxing, we won't need to repeat the full formatting block.